File: VB/EmployeeReport.vb
Imports DevExpress.XtraReports.UI
Imports System
Imports System.Collections
Imports System.ComponentModel
Imports System.Drawing

Namespace PassDataFromViewModelToReport
	Partial Public Class EmployeeReport
		Inherits DevExpress.XtraReports.UI.XtraReport

		Public Sub New()
			InitializeComponent()
		End Sub

		Public Sub New(ByVal dataSource As Object)
			Me.New()
			objectDataSource1.DataSource = dataSource
		End Sub
	End Class
End Namespace

File: CS/EmployeeReport.cs
using DevExpress.XtraReports.UI;
using System;
using System.Collections;
using System.ComponentModel;
using System.Drawing;

namespace PassDataFromViewModelToReport {
    public partial class EmployeeReport : DevExpress.XtraReports.UI.XtraReport {
        public EmployeeReport() {
            InitializeComponent();
        }

        public EmployeeReport(object dataSource) : this() {
            objectDataSource1.DataSource = dataSource;
        }
    }
}
